crates/ruma-identifiers/src/matrix_uri.rs

```diff
@@ -28,35 +28,112 @@ const TO_ENCODE: &AsciiSet = &CONTROLS
     .add(b';')
     .add(b'=');
 
-/// A reference to a user, room or event.
+/// All Matrix Identifiers that can be represented as a Matrix URI.
+#[derive(Debug, PartialEq, Eq)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub enum MatrixId {
+    /// A room ID.
+    Room(Box<RoomId>),
+
+    /// A room alias.
+    RoomAlias(Box<RoomAliasId>),
+
+    /// A user ID.
+    User(Box<UserId>),
+
+    /// An event ID.
+    Event { room_id: Box<RoomOrAliasId>, event_id: Box<EventId> },
```

Comment:
I forgot that it can be either a room ID or a room alias ID. In that case maybe it should be a tuple variant so we don't need to make the name longer for it to be correct? 😅